dynamodb-enhanced/runtime/src/main/java/io/quarkus/amazon/dynamodb/enhanced/runtime/DynamodbEnhancedClientRecorder.java

```diff
@@ -1,37 +1,96 @@
 import io.quarkus.arc.runtime.BeanContainerListener;
+import io.quarkus.runtime.RuntimeValue;
 import io.quarkus.runtime.annotations.Recorder;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbEnhancedClientExtension;
+import software.amazon.awssdk.enhanced.dynamodb.internal.client.ExtensionResolver;
 import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient;
 import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
 
 @Recorder
 public class DynamodbEnhancedClientRecorder {
+    private static final Log LOG = LogFactory.getLog(DynamodbEnhancedClientRecorder.class);
 
-    public BeanContainerListener setDynamoDbClient() {
+    DynamoDbEnhancedBuildTimeConfig buildTimeConfig;
+
+    public DynamodbEnhancedClientRecorder(DynamoDbEnhancedBuildTimeConfig buildTimeConfig) {
+        this.buildTimeConfig = buildTimeConfig;
+    }
+
+    public BeanContainerListener setDynamoDbClient(RuntimeValue<List<DynamoDbEnhancedClientExtension>> extensions) {
         BeanContainerListener beanContainerListener = new BeanContainerListener() {
 
             @Override
             public void created(BeanContainer container) {
                 DynamodbEnhancedClientProducer producer = container.beanInstance(DynamodbEnhancedClientProducer.class);
-                producer.setDynamoDbClient(container.beanInstance(DynamoDbClient.class));
+                producer.setDynamoDbClient(container.beanInstance(DynamoDbClient.class), extensions.getValue());
             }
         };
 
         return beanContainerListener;
     }
 
-    public BeanContainerListener setDynamoDbAsyncClient() {
+    public BeanContainerListener setDynamoDbAsyncClient(RuntimeValue<List<DynamoDbEnhancedClientExtension>> extensions) {
         BeanContainerListener beanContainerListener = new BeanContainerListener() {
 
             @Override
             public void created(BeanContainer container) {
                 DynamodbEnhancedClientProducer producer = container.beanInstance(DynamodbEnhancedClientProducer.class);
-                producer.setDynamoDbAsyncClient(container.beanInstance(DynamoDbAsyncClient.class));
+                producer.setDynamoDbAsyncClient(container.beanInstance(DynamoDbAsyncClient.class), extensions.getValue());
             }
         };
 
         return beanContainerListener;
     }
+
+    public RuntimeValue<List<DynamoDbEnhancedClientExtension>> createExtensionList() {
+
+        List<DynamoDbEnhancedClientExtension> extensions = new ArrayList<>();
+        for (String item : buildTimeConfig.clientExtensions.orElse(Collections.emptyList())) {
+            DynamoDbEnhancedClientExtension extension = createExtension(item.trim());
+            if (Objects.nonNull(extension)) {
+                extensions.add(extension);
+            }
+        }
+        if (extensions.isEmpty()) {
+            extensions.addAll(ExtensionResolver.defaultExtensions());
+        }
+        return new RuntimeValue<>(extensions);
```

Comment:
You can use ExtensionResolver.resolveExtension to reduce the list to one Extension and get rid of the list.

Reply:
Thanks, no I got it!